pennylane/transforms/optimization/pattern_matching.py

```diff
@@ -161,6 +161,8 @@ def circuit():
     3: ─├C─│───H─╰X─╰C─┤
     4: ─╰C─╰X──────────┤
 
+    .. seealso:: :func:`~.pattern_matching`
```

Comment:
Note: changes from https://github.com/PennyLaneAI/pennylane/pull/2489 have been directly incorporated in this PR.